pkg/k8sutils/redis-sentinel_test.go

```diff
@@ -384,8 +423,25 @@ func Test_getSentinelEnvVariable(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			ctx := context.TODO()
-			if got := getSentinelEnvVariable(ctx, tt.args.client, tt.args.cr, fake.NewSimpleDynamicClient(&runtime.Scheme{})); !reflect.DeepEqual(got, tt.want) {
+			dynamicClient := fake.NewSimpleDynamicClient(
+				runtime.NewScheme(),
+				&unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "redis.redis.opstreelabs.in/v1beta2",
+						"kind":       "RedisReplication",
+						"metadata": map[string]interface{}{
+							"name":      "redis-replication",
+							"namespace": "redis",
+						},
+						"spec": map[string]interface{}{
+							"clusterSize": int64(1),
+						},
+					},
+				},
+			)
+			if got, err := getSentinelEnvVariable(ctx, tt.args.client, tt.args.cr, dynamicClient); !reflect.DeepEqual(got, tt.want) {
```

Comment:
the `getSentinelEnvVariable` is calling `getRedisReplicationMasterPod` function that using redis client (it's looking for response of "master"), but it never happens, 

I was trying to add some client fake, but not sure how to go further, please check @drivebyer @shubham-cmyk

Reply:
I introduce gomonkey to fix it